src/cascadia/TerminalSettingsModel/AzureCloudShellGenerator.cpp

```diff
@@ -14,6 +14,8 @@ using namespace ::Microsoft::Terminal::Settings::Model;
 using namespace winrt::Microsoft::Terminal::Settings::Model;
 using namespace winrt::Microsoft::Terminal::TerminalConnection;
 
+std::wstring_view PROFILE_ICON_PATH_HIGH_RES{ L"ms-appx:///ProfileIcons/{b453ae62-4e3d-5e58-b989-0a998ec441b8}.scale-200.png" };

```

Comment:
please don't hardcode this.

the resource loader is supposed to handle this. if it's not, we need to investigate why THAT is.

if we have to hardcode `.scale-200` in the PNG filename... that means somebody is ignoring the actual scale factor.

This is important because we rely on the resource loader to handle ALL of the different resource qualifiers for us. if we had a pwsh.contrast-light.png file, we would not want to hardcode THAT either, right?

Reply:
Ok, removed the hardcoded reference. I decided to be more specific and created a new folder ProfileGeneratorIcons where I put the 32x32 icons of each generator there.

Admittedly, this was basically me copying over some of `.scale-200` PNG files into another folder which I'm not proud of haha. But, the images were specifically getting scaled because of the hardcoded Width/Height of 32 in the XAML. Didn't help if I changed the `IconSourceElement` into an `Image` to apply the `Stretch` property because at the end of the day, the resource loader was going to retrieve the `.scale-100` versions (which is fair because that would allow it to get a different scaling if necessary).

Hope this works, but let me know if you want any changes or if I misunderstood something! 😊 Resolving thread